addons/gsri_opex/functions/fremm/fn_fremmAddHeli.sqf

```diff
@@ -1,133 +1,129 @@
-};
-private _condition = {
-	// params ["_target", "_player", "_args"];
-	params ["", "", "_args"];
-	_args params ["_hangar"];
-	!isNull ([_hangar] call GSRI_fnc_heliRetrieveCurrent)
-};
-_actionList pushBack (["actionClear","Supprimer","",GSRI_fnc_heliRemove,_condition,{},[(_ship getVariable "GSRI_FREMM_hangar")],"",2,[false, false, false, false, false], _modifier] call ace_interact_menu_fnc_createAction);
+	// Adding fuel pomp
+	private _fuel = "Land_FuelStation_01_pump_malevil_F" createVehicle [0,0,0];
+	_fuel attachTo [_ship, [9.63,40.13,10.15]];
+	_fuel setDir 90;
 
-// FRIES mounting action
-private _condition = {
-	// params ["_target", "_player", "_args"];
-	params ["", "", "_args"];
-	_args params ["_hangar"];
-	private _heli = [_hangar] call GSRI_fnc_heliRetrieveCurrent;
-	(isNumber (configFile >> "CfgVehicles" >> typeOf _heli >> "ace_fastroping_enabled") && isNull (_heli getVariable ["ace_fastroping_FRIES", objNull]));
+	private _types = getArray (configFile >> "GSRI_FREMM_Templates" >> (_ship getVariable "GSRI_FREMM_selectTemplate") >> "AvailableHelis" >> "list");
+	{
+		private _helipad = "Land_HelipadEmpty_F" createVehicle [0,0,0];
+		_helipad attachTo [_ship, _x select 1];
+		_helipad setDir 180;
+		_helipad setVariable ["GSRI_FREMM_heli_list", _types, true];
+		_helipad setVariable ["GSRI_FREMM_associatedShip", _ship, true];
+		_ship setVariable [_x select 0, _helipad];
+	} forEach [["GSRI_FREMM_hangar", [0,44,8.81]], ["GSRI_FREMM_deck", [0.08,75,8.76]]];
 };
-_actionList pushBack (["actionFRIES",localize "STR_GSRI_FREMM_heliEquipFRIES","",GSRI_fnc_heliEquipFRIES,_condition,{},[(_ship getVariable "GSRI_FREMM_hangar")]] call ace_interact_menu_fnc_createAction);
 
-private _leshLoaded = isClass (configFile >> "CfgPatches" >> "rksla3_aircraft_tug");
-if(_leshLoaded && isServer) then {
-	// add RKSL airport tug to ship
-	private _tug = createVehicle ["rksla3_aircraft_tug_blufor", [0,0,100], [], 0, "NONE"];
-	_tug enableSimulation false;
-	_tug allowDamage false;
-	_tug animateSource ["option_cabin_hide_source", 0, true];
-	_tug setDir getDir _ship;
-	_tug setPosATL ASLToATL (_ship modelToWorldWorld [7.45,47.77,8.81]);
-	_tug enableSimulation true;
-	_tug allowDamage true;
-	_ship setVariable ["GSRI_FREMM_tug", _tug, true];
-} else {
-	// Add placeholder actions
-	// Place helicopter on rear deck
-	// Will not be externalized
-	private _condition = {
-		// params["_target", "_player", "_args"];
-		params["", "", "_args"];
-		_args params ["_ship"];
-		// Return true if there is a heli in hangar and if there is no heli on deck
-		!(isNull ([(_ship getVariable "GSRI_FREMM_hangar")] call GSRI_fnc_heliRetrieveCurrent)) and (isNull ([(_ship getVariable "GSRI_FREMM_deck")] call GSRI_fnc_heliRetrieveCurrent))
+// Clientside-only : basically ACE actions
+if!(isDedicated) then {
+	// Actions handle spawn
+	private _handle = ((_ship getVariable "GSRI_FREMM_screens") select 0);
```

Comment:
don't we have a function for this in OPEX ?

Reply:
Thinking of this https://github.com/team-gsri/GSRI-Opex/blob/master/addons/gsri_opex/functions/screen/fn_screenGetById.sqf i guess ? I'll switch to that when back to home